### ISSUE TITLE
CUDA JIT launch configuration failure at large sizes

### ISSUE THREAD

**Issue by 9prady9** [arrayfire/arrayfire#1894]
Reported on Google forums by Aaron W. Hsu

Code used to reproduce this. The issue is present in CUDA backend alone. CPU/OpenCL backends are working fine. Also, If I bring the size down low enough (1024 * 1024) then it works, and if I use a constant() form instead of a singleton mutable array value, it also works.
```cpp
#include <iostream>
#include <vector>
#include <arrayfire.h>
 
using namespace af;
 
int main(int argc, char *argv[])
{
  dim_t size = 2 * 1024 * 1024; // 1024 * 1024 works
  array a1 = randu(size,f64);
  /* array a2 = constant(5, 1, f64); // Works */
  array a2 = array(1, f64);
  a2(0) = 5;
  array a3 = tile(a2, size);
  std::vector<double> x(size);
 
  try {
          array a4 = a1*a3;
          a4.host(x.data());
  }
  catch (exception x) {
          std::cout << x << std::endl;
  }
 
  return 0;
}
```

Error
```
a3 dims 2097152 1 1 1
a4 dims 2097152 1 1 1
In function void cuda::evalNodes(std::vector<cuda::Param<T> >&, std::vector<cuda::JIT::Node*>) [with T = double]
In file src/backend/cuda/jit.cpp:437
CU Error (1)


ArrayFire Exception (Internal error:998):
In function void cuda::evalNodes(std::vector<cuda::Param<T> >&, std::vector<cuda::JIT::Node*>) [with T = double]
In file src/backend/cuda/jit.cpp:437
CU Error (1)


In function void af::array::host(void*) const
In file src/api/cpp/array.cpp:254

```

I was able to reproduce this on hotifxes branch on my end as well.

**Comment by pavanky** [comment]
being fixed in https://github.com/arrayfire/arrayfire/pull/1893

**Comment by 9prady9** [comment]
@pavanky You have more commits coming in for that PR ?

**Comment by pavanky** [comment]
@9prady9 yes fixing this very case. Apparently my assumption that drvier api and runtime api have same launch limitations was wrong.

**Comment by pavanky** [comment]
Shortest way to reproduce the same issue:

```
TEST(JIT, ISSUE_1894)
{   
    af::array a = af::randu(1);
    af::array b = af::tile(a, 2 * (1 << 20));
    af::eval(b);
}
```

**Comment by pavanky** [comment]
@9prady9 check now.

**Comment by 9prady9** [comment]
@pavanky Awesome, that worked.

**Comment by pavanky** [comment]
Fixed in hotfixes-3.5.1
